TESTS/mbed_hal/lp_us_tickers/main.cpp

```diff
@@ -13,98 +13,416 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#include "utest/utest.h"
-#include "unity/unity.h"
+#include "mbed.h"
 #include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+#include "ticker_api_tests.h"
+#include "hal/us_ticker_api.h"
+#include "hal/lp_ticker_api.h"
 
-#include "mbed.h"
-#include "us_ticker_api.h"
-#include "ticker_api.h"
+#define FORCE_OVERFLOW_TEST (false)
+#define TICKER_INT_VAL 5000
+#define TICKER_DELTA 50
+
+#if DEVICE_LOWPOWERTIMER
+#define TICKER_OVERFLOW_DELTA 0 // this will allow to detect that ticker counter rollovers to 0
+#else
+#define TICKER_OVERFLOW_DELTA 50
+#endif
+
+#define TICKER_100_TICKS 100
+
+#define MAX_FUNC_EXEC_TIME_US 20
+#define DELTA_FUNC_EXEC_TIME_US 5
+#define NUM_OF_CALLS 1000
+
+#define NUM_OF_CYCLES 100000
 
 using namespace utest::v1;
 
-namespace {
-    volatile bool complete;
-    const ticker_interface_t* intf;
+volatile int intFlag = 0;
+const ticker_interface_t* intf;
+
+/* Auxiliary function to count ticker ticks elapsed during execution of N cycles of empty while loop.
+ * Parameter <step> is used to disable compiler optimisation. */
+uint32_t count_ticks(uint32_t cycles, uint32_t step)
```

Comment:
What about making cycles volatile to prevent compiler optimization

Reply:
fixed.